app/models/package_manager/pypi.rb

```diff
@@ -123,14 +125,22 @@ def self.known_versions(name)
         &.index_by { |v| v[:number] } || {}
     end
 
+    # Simply parses out the name of a PEP 508 Dependency specification: https://peps.python.org/pep-0508/
+    # Leaves the rest as-is with any leading semicolons or spaces stripped
+    def self.parse_pep_508_dep_spec(dep)
+      name, requirement = dep.split(PYPI_508_NAME_REGEX, 2).last(2).map(&:strip)
```

Comment:
is `last(2)` needed here? and curious if it is, wouldn't we want the `first(2)`?

Reply:
when we split on the name, what we get is a blank string (what's to the left of the name) and the requirements (to the right). Because we have a capture group on the name itself, apparently the `split` method doesn't count the captured matches towards the _limit_ parameter. https://apidock.com/ruby/String/split
> If limit is a positive number, at most that number of [split](https://apidock.com/ruby/String/split) substrings will be returned (captured groups will be returned as well, but are not counted towards the limit)

Without the capture group and with a limit of 2 we get `[blank string, requirements]`, but with the capture group, it seems to stick it in the middle so `[blank string, name, requirements]`. 

In stepping through and testing this again I did discover we don't need the part on the line after which defaults requirement to `""` if it's `nil`. After fixing the regex to cover single character packages it now has `[blank string, name, blank string]` for those packages.

Reply:
woah, that's a little odd but makes sense I guess.